src/helpers/buildPaperslistByContact.js

```diff
@@ -1,45 +1,67 @@
 import { models } from 'cozy-client'
-import { isReferencedById } from 'cozy-client'
+import { isReferencedById, isReferencedBy } from 'cozy-client'
 import { CONTACTS_DOCTYPE } from 'src/doctypes'
 const { getDisplayName } = models.contact
 
+const getPaperWithoutContact = papersList => {
+  return papersList.filter(paper => !isReferencedBy(paper, CONTACTS_DOCTYPE))
+}
+
 const DEFAULT_MAX_DISPLAY = 3
 
 /**
  * @typedef {object} BuildPaperslistByContactParam
  * @property {object[]} papersList - Array of IOCozyFile
  * @property {object[]} contactsList - Array of cozy contact object
+ * @property {object[]} defaultName - Name of header of papers without contact
  * @property {Paper[]} [papersDefinitions=[]] Array of Papers
  * @property {string} [currentFileCategory=''] - Category of qualification
  */
 
 /**
  * @param {BuildPaperslistByContactParam} param
- * @returns {{ contact: string, papers: {maxDisplay: number, list: IOCozyFile[]} }[]}
+ * @returns {{ withHeader: boolean, contact: string, papers: {maxDisplay: number, list: IOCozyFile[]} }[]}
  */
 export const buildPaperslistByContact = ({
   papersList,
   contactsList,
+  defaultName,
   papersDefinitions = [],
   currentFileCategory = ''
 }) => {
   let result = []
-  if (Array.isArray(contactsList) && Array.isArray(papersList)) {
-    const currentDef = papersDefinitions.find(
-      paperDef => paperDef.label === currentFileCategory
-    )
-    result = contactsList.flatMap(contact => [
-      {
-        contact: getDisplayName(contact),
-        papers: {
-          maxDisplay: currentDef?.maxDisplay || DEFAULT_MAX_DISPLAY,
-          list: papersList.filter(paper =>
-            isReferencedById(paper, CONTACTS_DOCTYPE, contact._id)
-          )
-        }
+  const paperWithoutContact = getPaperWithoutContact(papersList)
+  const currentDef = papersDefinitions.find(
+    paperDef => paperDef.label === currentFileCategory
+  )
+  const maxDisplay = currentDef?.maxDisplay || DEFAULT_MAX_DISPLAY
+  const withHeader = !(
+    paperWithoutContact.length === papersList.length && papersList.length > 0
+  )
+
+  result = contactsList.map(contact => ({
+    withHeader,
+    contact: getDisplayName(contact),
+    papers: {
+      maxDisplay,
+      list: papersList.filter(paper =>
+        isReferencedById(paper, CONTACTS_DOCTYPE, contact._id)
+      )
+    }
+  }))
+
+  const resultSorted = result.sort((a, b) => (a.contact > b.contact ? 1 : -1))
```

Comment:
Since you can sort non-ascii character, you should use localCompare instead of just `>` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#sorting_non-ascii_characters